src/pip/_internal/operations/prepare.py

```diff
@@ -38,6 +46,19 @@ def make_abstract_dist(req):
         return IsSDist(req)
 
 
+def _install_build_reqs(finder, prefix, build_requirements):
+    finder = copy(finder)
+    finder.format_control = FormatControl(set(), set([":all:"]))
+    urls = [finder.find_requirement(InstallRequirement.from_line(r),
```

Comment:
The whole point of linting is to make it easier to read stuff; ignoring linters for and doing a review is something I'd like to do avoid in the future.